### ISSUE TITLE
Environment tries to add player while updating causing a ConcurrentModificationException.

### ISSUE THREAD

**Issue by Jidek** [nashkevin/EECS-393-Project#6]
When the environment updates it is iterating through the hash sets of game objects. 
When a player enters the game their game object is added to the appropriate hash set. 

If both of these things happen at the same time then the following error gets thrown:
`Exception in thread "Thread-14" java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1437)
	at java.util.HashMap$KeyIterator.next(HashMap.java:1461)
	at main.java.environment.Environment.update(Environment.java:126)
	at main.java.environment.Environment.run(Environment.java:138)`

**Comment by nashkevin** [comment]
Should be fixed as of b5a1eea57ea382d2b6c0be665ed9e91c04a8255b, but we should confirm with testing.

**Comment by nashkevin** [comment]
The following pops up occasionally when running `mvn test`. This time the exception was thrown when WebServer is trying to broadcast.
```
java.util.ConcurrentModificationExceptionMessage from 55e86061-0e47-4062-af14-29
0f12e2ff58:  {"name":"Test2"}

        at java.util.HashMap$HashIterator.nextNode(HashMap.java:1437)
        at java.util.HashMap$KeyIterator.next(HashMap.java:1461)
        at main.java.web.WebServer.broadcast(WebServer.java:97)
        at main.java.web.WebServer.broadcast(WebServer.java:105)
        at main.java.web.GameThread.run(GameThread.java:51)
```
